q2galaxy/core/usage.py

```diff
@@ -105,15 +108,50 @@ def to_interface_name(self, skip_ref=False):
             return '.'.join([self.prefix, name])
         return name
 
-    def assert_output_type(self, semantic_type):
+    def assert_output_type(self, semantic_type, key=None):
         semantic_type = re.escape(str(semantic_type))
-        self._galaxy_has_line_matching(path='metadata.yaml',
+        path = 'metadata.yaml'
+
+        if key is not None:
+            key = str(key)
+
+        if self.var_type == 'result_collection' and key:
+            self._key_helper(path, expression='type: EchoOutput', key=key)
+            return
```

Comment:
This isn't quite right. We need to use `semantic_type` here still, it's just the path we are talking about becomes different. So `_galaxy_has_line_matching` needs to index into the collection.

Reply:
I see what `_key_helper` is kind of doing. It probably isn't the best name here.